app/models/atmosphere/virtual_machine_flavor.rb

```diff
@@ -60,6 +66,37 @@ class VirtualMachineFlavor < ActiveRecord::Base
 
     scope :active, -> { where(active: true) }
 
+    # Assumes only 1 vmf_osf for a specific os_family
+    def get_hourly_cost_for(os_family)
+      incarnation = virtual_machine_flavor_os_families.find_by(os_family: os_family)
+      incarnation && incarnation.hourly_cost # nil if incarnation.blank?
+    end
+
+    # Upserts a binding between this VMF and an OS family, setting hourly cost
+    def set_hourly_cost_for(os_family, cost)
+      incarnation = virtual_machine_flavor_os_families.find_by(os_family: os_family)
+      if incarnation.blank?
+        incarnation = VirtualMachineFlavorOSFamily.new
+        incarnation.os_family = os_family
+        incarnation.virtual_machine_flavor = self
```

Comment:
Please use reference for creation or object initialization.

```ruby
virtual_machine_flavor_os_families.initialize_by(os_family: os_family)
```

When you are using this method than `self` is aware about new object in relation otherwise (like in your current version) it doesn't.